R/bench-auroc.R

```diff
@@ -0,0 +1,132 @@
+#' This function takes the elements of the `activity` column and calculates
+#' precision-recall and ROC curves (depending on `curve`).
+#' The `activity` column is populated with the output for each stat method and
+#' results from the `run_benchmark()` function. Each of the elements
+#' in `activity` are the results from runs of the `decouple()` wrapper.
+#'
+#' @param df run_benchmark roc column provided as input
+#' @param downsampling logical flag indicating if the number of TN should be
+#'   downsampled to the number of TP
+#' @param times integer showing the number of downsampling
+#' @param ranked logical flag indicating if input is derived from composite
+#'   ranking that already took up-/downregulation (sign) into account
+#' @param curve whether to return a Precision-Recall Curve ("PR") or ROC ("ROC")
+#'
+#' @return tidy data frame with precision, recall, auc, n, tp, tn and coverage
+#'
+#' @import yardstick
+calc_curve = function(df, downsampling = T,
```

Comment:
Also, style guidelines consider setting `T` or `F` as a bad practice in boolean context since you can create variables with those names. Please change it to `TRUE` or `FALSE` instead.